controllers/controllers/cluster_controller_test.go

```diff
@@ -0,0 +1,378 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/networkutils"
+	"github.com/aws/eks-anywhere/pkg/providers/vsphere"
+	"github.com/aws/eks-anywhere/pkg/providers/vsphere/mocks"
+	"github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+var (
+	name      = "test-cluster"
+	namespace = "default"
+)
+
+func TestClusterReconcilerSkipManagement(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	govcClient := mocks.NewMockProviderGovcClient(ctrl)
+
+	secret := createSecret()
+	cluster := createCluster()
+	datacenterConfig := createDataCenter(cluster)
+	bundle := createBundle(cluster)
+	machineConfigCP := createCPMachineConfig()
+	machineConfigWN := createWNMachineConfig()
+
+	objs := []runtime.Object{cluster, datacenterConfig, secret, bundle, machineConfigCP, machineConfigWN}
+
+	s := scheme.Scheme
+	s.AddKnownTypes(anywherev1.GroupVersion, cluster)
+	s.AddKnownTypes(anywherev1.GroupVersion, datacenterConfig)
+	s.AddKnownTypes(anywherev1.GroupVersion, bundle)
+	s.AddKnownTypes(anywherev1.GroupVersion, machineConfigCP)
```

Comment:
nit: we can avoid this by `import _ "github.com/aws/eks-anywhere/internal/test/envtest"`

Reply:
done